src/wallet.rs

```diff
@@ -280,37 +289,42 @@ impl Wallet {
     self.settings.integration_test()
   }
 
-  pub(crate) fn is_mature(&self, rune: Rune, commit: &Transaction) -> Result<bool> {
-    let transaction = self
-      .bitcoin_client()
-      .get_transaction(&commit.txid(), Some(true))
-      .into_option()?;
+  fn is_after_minimum_height(&self, rune: Rune) -> bool {
+    rune
+      >= Rune::minimum_at_height(
+        self.chain().network(),
+        Height(u32::try_from(self.bitcoin_client().get_block_count().unwrap() + 1).unwrap()),
+      )
+  }
 
-    if let Some(transaction) = transaction {
-      if u32::try_from(transaction.info.confirmations).unwrap() + 1
-        >= Runestone::COMMIT_CONFIRMATIONS.into()
-        && rune
-          >= Rune::minimum_at_height(
-            self.chain().network(),
-            Height(u32::try_from(self.bitcoin_client().get_block_count()? + 1).unwrap()),
-          )
+  pub(crate) fn is_mature(&self, rune: Rune, commit: &Transaction) -> Result<Maturity> {
```

Comment:
Should we change the method name too here @raphjaph ? Now that it doesn't return bool, `is_mature` is less suitable here. We can name it `check_maturity` or something to reflect enum return type.

Reply:
good point, will change